contentcuration/contentcuration/frontend/shared/feedbackApiUtils.js

```diff
@@ -0,0 +1,148 @@
+// Helper functions and Utils for creating an API request to
+// Feedback mechanism endpoints
+import { v4 as uuidv4 } from 'uuid';
+import client from './client';
+
+export const FeedbackTypeOptions = {
+  imported: 'imported',
+  rejected: 'rejected',
+  previewd: 'previewd',
+  showmore: 'showmore',
+  ignored: 'ignored',
+  flagged: 'flagged',
+};
+
+// This is mock currently, fixed value of URL still to be decided
+export const FLAG_FEEDBACK_EVENT_URL = '/api/flagged/';
+
+/**
+ * Base class for feedback data objects.
+ */
+class BaseFeedback {
+  /**
+   * Initializes a new BaseFeedback object.
+   *
+   * @class
+   * @classdesc Represents a base feedback object with common properties and methods.
+   *
+   * @param {Object} object - Object passed down following the .
+   * @param {Object} [object.context={}] - The context associated with the ObjectType.
+   * @param {string} object.contentnode_id - Id for ContentNode that does not change.
+   * @param {string} object.content_id - content_id of ContentNode wrt to ObjectType.
+   */
+  constructor({ context = {}, contentnode_id, content_id }) {
+    this.id = uuidv4();
+    this.context = context;
+    this.contentnode_id = contentnode_id;
+    this.content_id = content_id;
+  }
+
+  // Creates a data object according to Backends expectation,
+  // excluding functions and the "URL" property.
+  getDataObject() {
+    const dataObject = {};
+    for (const key in this) {
+      if (
+        Object.prototype.hasOwnProperty.call(this, key) &&
+        typeof this[key] !== 'function' &&
+        key !== 'URL'
+      ) {
+        dataObject[key] = this[key];
+      }
+    }
+    return dataObject;
+  }
+
+  // Return URL associated with the ObjectType
+  getUrl() {
+    if (this.defaultURL === null || this.URL === undefined) {
+      throw new Error('URL is not defined for the FeedBack Object.');
+    }
+    return this.URL;
+  }
+}
+
+/**
+ * Initializes a new BaseFeedbackEvent object.
+ *
+ * @param {Object} params - Parameters for initializing the object.
+ * @param {Object} params.user - The user associated with the feedback event.
+ * @param {string} params.target_channel_id - The ID of the target channel for the feedback event.
+ * @param {Object} baseFeedbackParams - Additional parameters inherited from the base feedbackclass.
+ */
+// eslint-disable-next-line no-unused-vars
```

Comment:
What was eslint complaining about here?

Reply:
Currently the BaseFeedbackEvent is not being used anywhere as the recommendations core classes like ReccommendationInteractionEvent and ReccommendationEvent class  are still to be added! Hence its suggesting to remove the unused class.

Reply:
So this file is treated as an ES module, meaning if variables aren't used in the file then this would occur. Nothing is exported from this module, so doing that could perhaps resolve it?